src/compiler/scanner.ts

```diff
@@ -2623,11 +2635,11 @@ export function createScanner(languageVersion: ScriptTarget,
         languageVariant = variant;
     }
 
-    function setTextPos(textPos: number) {
-        Debug.assert(textPos >= 0);
-        pos = textPos;
-        startPos = textPos;
-        tokenPos = textPos;
+    function resetTokenState(position: number) {

```

Comment:
The fact that this hanles positions but talks about token state feels off.

Reply:
`resetTokenPositions` ? `resetTokenToPosition` ? `resetTokenAtPosition` ? But it also resets token kind, flags and value.